tests/main/snapctl/task.yaml

```diff
@@ -55,12 +55,20 @@ restore: |
 
 execute: |
     echo "Run the hook that calls snapctl"
-    export SNAP_CONTEXT="foo" # Just enough to hit the API
-    if ! snap run --hook=apply-config snapctl-hooks; then
-        echo "Failed to run hook"
+
+    # This context is enough to hit the API, but the hook will fail since this
+    # context is obviously invalid. That failure still means we're correctly
+    # hitting the API, though, which is all we care about here.
+    export SNAP_CONTEXT="foo"
+    if output="$(snap run --hook=apply-config snapctl-hooks)"; then
+        echo "Expected the hook to fail"
         exit 1
     fi
 
+    if [[ ! "$output" =~ .*"no context for ID: \"foo\"".* ]]; then
```

Comment:
Or alternatively, using the bash expression syntax, but simply:

`[[ "$output" =~ 'no context for ID: "foo"' ]]`

I've just confirmed that it traces properly as well, and actually better than echo+grep.